lib/bitbucket/bitbucket.go

```diff
@@ -19,13 +19,17 @@
 package bitbucket
 
 import (
+	"github.com/zitadel/oidc/v3/pkg/oidc"
+
 	workloadidentityv1pb "github.com/gravitational/teleport/api/gen/proto/go/teleport/workloadidentity/v1"
 )
 
 // IDTokenClaims
 // See the following for the structure:
 // https://support.atlassian.com/bitbucket-cloud/docs/integrate-pipelines-with-resource-servers-using-oidc/
 type IDTokenClaims struct {
+	oidc.Claims
```

Comment:
Iirc - if you have a sub field on the custom struct, you need to implement GetSubject() for Zitadel OIDC to be able to check the subject properly as the custom struct will take priority during unmarshaling.